scripts/bad_words.py

```diff
@@ -0,0 +1,67 @@
+"""
+
+Usage:
+    scripts/bad_words.py <data_api_url> <data_api_token> <bad_words_path> <framework_slug>
+"""
+from docopt import docopt
+from dmutils.apiclient import DataAPIClient
+import six
+import re
+
+
+def main(data_api_url, data_api_token, bad_words_path, framework_slug):
+    client = DataAPIClient(data_api_url, data_api_token)
+    bad_words = get_bad_words(bad_words_path)
+    suppliers = get_suppliers(client, framework_slug)
+    get_services_with_bad_words(client, suppliers, bad_words)
+
+
+def get_suppliers(client, framework_slug):
+    suppliers = client.find_framework_suppliers(framework_slug)
+    suppliers = suppliers["supplierFrameworks"]
+    if (framework_slug == "g-cloud-6"):
+        suppliers_on_framework = suppliers
+    else:
+        suppliers_on_framework = [supplier for supplier in suppliers if supplier["onFramework"]]
+    return suppliers_on_framework
+
+
+def get_draft_services(client, supplier_id, framework_slug):
+    services = client.find_draft_services(supplier_id, framework=framework_slug)
+    services = services["services"]
+    submitted_services = [service for service in services if service["status"] == "submitted"]
+    return submitted_services
+
+
+def get_services(client, supplier_id, framework_slug):
+    services = client.find_services(supplier_id, framework=framework_slug)
+    services = services["services"]
+    return services
+
+
+def get_bad_words(bad_words_path):
+    with open(bad_words_path) as file:
+        lines = file.readlines()
+        return [line.strip() for line in lines if not (line.startswith("#") or line.isspace())]
+
+
+def get_services_with_bad_words(client, suppliers, bad_words):
+    for supplier in suppliers:
+        if supplier["frameworkSlug"] == "g-cloud-6":
+            services = get_services(client, supplier["supplierId"], supplier["frameworkSlug"])
+        else:
+            services = get_draft_services(client, supplier["supplierId"], supplier["frameworkSlug"])
+        for service in services:
+            for key in service:
+                if isinstance(service[key], six.string_types):
```

Comment:
This will miss keys like `serviceFeatures`, which are lists of user-submitted strings.